### ISSUE TITLE
Consider how to improve suggestions refresh

### ISSUE THREAD

**Issue by semanticer** [toggl/android#466]
## Description
This relates to the discussion we had with @daividssilverio about https://github.com/toggl/android/pull/434 more precisely this [line
](https://github.com/toggl/android/pull/434/files#diff-659b1729e8ea5eadd4d75817ba826017R65) 
We should consider either less "random" place to dispatch these kind of actions or the suggestion providers should be more reactive and should push changes whenever there are some.

## Dependencies
none

## Definition of done
- [ ] New clear place for these fetch actions is defined
or
- [ ] Subscriptions are updated reactively and we don't need this perpetual fetch anymore.

**Comment by heytherewill** [comment]
Let's revisit this after #329

**Comment by semanticer** [comment]
#329 is merged and subscriptions should be the answer here for sure. 
Here is the new issue https://github.com/toggl/android/issues/639 and I'm closing this one.
